packages/sign-client/src/controllers/engine.ts

```diff
@@ -434,9 +433,15 @@ export class Engine extends IEngine {
         params: getSdkError("USER_DISCONNECTED"),
         throwOnFailedPublish: true,
       });
-      await this.deleteSession(topic);
-    } else {
+      await this.deleteSession({ topic, emitEvent: false });
+    } else if (this.client.core.pairing.pairings.keys.includes(topic)) {
       await this.client.core.pairing.disconnect({ topic });
+    } else {
+      const { message } = getInternalError(
+        "MISMATCHED_TOPIC",
+        `No Session or Pairing topic found: ${topic}`,
```

Comment:
Nit: adjusting casing and aligning the wording to similar cases in the rest of the codebase

```suggestion
        `Session or pairing topic not found: ${topic}`,
```